app/src/main/java/org/mozilla/fenix/ext/BrowserIcons.kt

```diff
@@ -4,8 +4,35 @@
 
 package org.mozilla.fenix.ext
 
+import android.graphics.drawable.Drawable
+import android.view.View
 import android.widget.ImageView
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import mozilla.components.browser.icons.BrowserIcons
 import mozilla.components.browser.icons.IconRequest
 
-fun BrowserIcons.loadIntoView(view: ImageView, url: String) = loadIntoView(view, IconRequest(url))
+fun BrowserIcons.loadIntoView(
+    view: ImageView,
+    url: String = "",
+    localImage: Drawable? = null,
+    iconCover: ImageView? = null
+): Job? {
+    if (localImage != null) view.setImageDrawable(localImage)
```

Comment:
If we look at the [internals](https://searchfox.org/mozilla-mobile/source/android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt#181) of `loadIntoView`, we already have an API that sets a placeholder until the root image is available. Would that work?

Reply:
I have tried using it, but the refresh of HomeFragment when clicking new tab FAB would make the whole topsites area flicker somehow. I'll look into it some more.

Reply:
Maybe you're seeing https://github.com/mozilla-mobile/fenix/issues/15499 which I was struggling to fix as well. Might be time to fix this in that case if it's stopping up for building other features.